IdentityCore/src/requests/broker/MSIDSSOExtensionSilentTokenRequest.m

```diff
@@ -140,30 +150,49 @@ - (void)executeRequestWithCompletion:(MSIDRequestCompletionBlock)completionBlock
         NSDictionary *mamResources = [self.mamResourcesCache resourcesJsonDictionaryWithContext:self.requestParameters
                                                                                           error:nil];
         
-        __auto_type operationRequest = [MSIDBrokerOperationSilentTokenRequest tokenRequestWithParameters:self.requestParameters
+        self.operationRequest = [MSIDBrokerOperationSilentTokenRequest tokenRequestWithParameters:self.requestParameters
```

Comment:
I'm not sure how to measure it, the idea is later when receiving the response from SSO-ext I need that object (because it has the calculated thumbprint). So I move it from method object to class object, and later use it by:
                    [weakSelf.throttlingService updateThrottlingService:error tokenRequest:_**weakSelf.operationRequest**_];

Reply:
I was specifically worried about retain cycles. Basically, if MSIDBrokerOperationSilentTokenRequest strongly retains MSIDSSOExtensionSilentTokenRequest, and now MSIDSSOExtensionSilentTokenRequest strongly retains MSIDBrokerOperationSilentTokenRequest, then memory won't get cleaned up. You can check retain cycles in the instruments or Xcode memory diagnostics. Strong retain doesn't always mean a retained property to the object - it can be something less obvious like reference in the block, so it's good to double check.

Reply:
Sounds interesting. Noted it, will comeback with the result.

Reply:
https://microsoft-my.sharepoint-df.com/:i:/p/hiengu/EWzUCJq6hxRIqVYEQul7PNwBIanANr9tIJzJDTdn9qQSOA?e=FpcG6c
https://microsoft-my.sharepoint-df.com/:i:/p/hiengu/ESgw7taaTyJJrPLQgzdzPZIBj5TUUsfISBw0DqseXBT4Lw?e=QSNgc5
I just had a quick look at this. Looking at the RefCt and retain cycle of MSIDBrokerOperationSilentTokenRequest and MSIDSSOExtensionSilentTokenRequest I think they are both healthy, the RefCt eventually count to 0.
Let me know if that makes sense and we still have any concern on this @oldalton @NerevarineRule @kaisong1990